cyto_dl/image/transforms/maxpool_resize.py

```diff
@@ -0,0 +1,139 @@
+    tensor to the specified `spatial_size`. It supports:
+    - 3D tensors ([channels, height, width]) for single images.
+    - 4D tensors ([batch, channels, height, width]) for batched images.
+    - 5D tensors ([batch, channels, depth, height, width]) for volumetric data.
+
+    The transform uses `torch.nn.functional.adaptive_max_pool2d` for 2D data and
+    `adaptive_max_pool3d` for 3D data to ensure precise output sizes. Non-positive values
+    in `spatial_size` are replaced with the corresponding input dimensions.
+
+    Parameters
+    ----------
+    keys: Union[str, Sequence[str]]
+        Keys of the corresponding items to be transformed in the input dictionary.
+    spatial_size: Union[Sequence[int], int]
+        Expected shape of spatial dimensions after resize operation.
+        If a single integer is provided, it is applied to all spatial dimensions
+        (e.g., 32 -> (32, 32) for 2D or (32, 32, 32) for 3D).
+        If a sequence is provided, its length must match the number of spatial dimensions
+        (2 for images, 3 for volumes).
+    allow_missing_keys: bool, optional
+        If True, skips missing keys in the input data without raising an error.
+        Default is False.
+
+    Raises
+    ------
+    TypeError
+        If the input for a key is not a PyTorch tensor.
+    ValueError
+        If the input tensor has an unsupported number of dimensions, if `spatial_size`
+        has an invalid length, or if the output spatial size is invalid (e.g., larger
+        than the input size).
+    KeyError
+        If a key is missing in the input data and `allow_missing_keys` is False.
+ 
+    """
+
+    def __init__(
+        self,
+        keys: Union[str, Sequence[str]],
+        spatial_size: Union[Sequence[int], int],
+        allow_missing_keys: bool = False,
+    ):
+        super().__init__()
+        self.keys = keys if isinstance(keys, (list, ListConfig)) else [keys]
+        self.allow_missing_keys = allow_missing_keys
+
+        # Store raw spatial_size for later validation
+        self.spatial_size = spatial_size
+
+        # Validate spatial_size length in advance
+        if isinstance(spatial_size, (list, tuple, ListConfig)):
+            if len(spatial_size) not in (2, 3):
+                raise ValueError(
+                    f"spatial_size sequence must have length 2 or 3, got length {len(spatial_size)}"
+                )
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            if key not in d:
+                if self.allow_missing_keys:
+                    continue
+                raise KeyError(f"Key '{key}' not found in input data.")
+
+            x = d[key]
+            if not isinstance(x, torch.Tensor):
+                raise TypeError(f"Input '{key}' must be a PyTorch tensor, got {type(x)}")
+
+            # Determine expected tensor dimensions and spatial size length
+            input_dims = x.dim()
```

Comment:
Monai transforms expect C[Z]YX images - would it be reasonable to enforce this based on the spatial size and then just add the batch dimension right before pooling? I often have an explicit spatial_dims argument to help with this kind of check